test/integration/http2_integration_test.h

```diff
@@ -5,10 +5,10 @@
 #include "gtest/gtest.h"
 
 namespace Envoy {
-class Http2IntegrationTest : public BaseIntegrationTest,
+class Http2IntegrationTest : public HttpIntegrationTest,
```

Comment:
While we're thinking about refactoring, protocols and programmatic generation of config, what are your thoughts about being able to independently set the downstream/upstream protocol? Today, upstream is essentially a property of how we setup the config file, rather than test driven. It seems that (just like in GFE), we want to be able to treat these both as first class properties in tests.

Reply:
Yeah, I think that one may have to want until backend config is proto-driven, but I planned to experiment with it next.